### PR TITLE
Embcess 428 add audit fields to tables

### PR DESCRIPTION
**Pull request by CodingVelocista**

### CODE REVIEW

File: embc-app/DataInterfaces/EmbcDbContext.cs [hunk 1]
```diff
@@ -129,6 +131,23 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
                 .Property(r => r.EssFileNumber)
                 .HasDefaultValueSql("NEXT VALUE FOR ESSFileNumbers");
+
+            modelBuilder.ShadowProperties();
+
+        }
+
+        public override int SaveChanges()
+        {
+            ChangeTracker.SetShadowProperties(ctx?.HttpContext?.User);
```
Comment (ytqsl) [inline embc-app/DataInterfaces/EmbcDbContext.cs]: Why is the code in an extension method and not here? Is it a best practice?

File: embc-app/DataInterfaces/ModelBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
+using Gov.Jag.Embc.Public.Models.Db;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Gov.Jag.Embc.Public.DataInterfaces
+{
+    public static class ModelBuilderExtensions
```
Comment (ytqsl) [inline embc-app/DataInterfaces/ModelBuilderExtensions.cs]: Can we merge all extension methods to a single Ex file or merge into the context file instead? Is using extension methods a best practice for this?

File: embc-app/Models/Db/HeadOfHousehold.cs [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 namespace Gov.Jag.Embc.Public.Models.Db
 {
-    public abstract class EvacueeOld : Person
+    public abstract class EvacueeOld : Person, IAuditableEntity
```
Comment (ytqsl) [inline embc-app/Models/Db/HeadOfHousehold.cs]: Is this required? This entity will go away, why do we need to change it?

Comment (CodingVelocista) [reply embc-app/Models/Db/HeadOfHousehold.cs]: This was left over from the previous version, so it's not needed.  The will be dropped, once we move to TEST.

File: embc-app/DataInterfaces/ModelBuilderExtensions.cs (hunk 2 above)
Comment (CodingVelocista) [reply embc-app/DataInterfaces/ModelBuilderExtensions.cs]: We can definitely merge the the logic from the SaveChanges directly into the override.  As for the ModelBuilderExtensions, I was unsuccessful to get this functionality to work successfully outside of this pattern.

File: embc-app/DataInterfaces/EmbcDbContext.cs [hunk 4]
```diff
@@ -129,6 +136,43 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+        }
+
+        public override async Task<int> SaveChangesAsync(CancellationToken cancellationToken = new CancellationToken())
+        {
+            SetShadowProperties(ctx?.HttpContext?.User);
+
+            return await base.SaveChangesAsync(cancellationToken);
+        }
+
+        public void SetShadowProperties(ClaimsPrincipal principal)
```
Comment (ytqsl) [inline embc-app/DataInterfaces/EmbcDbContext.cs]: private?

File: embc-app/Models/Db/HeadOfHousehold.cs (hunk 3 above)
Comment (ytqsl) [reply embc-app/Models/Db/HeadOfHousehold.cs]: Do we want to add the interface on this entity?

**Pull request merged**